snapcraft/internal/meta/_desktop.py

```diff
@@ -28,28 +28,26 @@ class DesktopFile:
 
     def __init__(
             self, *, name: str, filename: str,
-            snap_name: str, prime_dir: str) -> None:
+            snap_name: str, prime_dir: str, app_name: str, section_name: str) -> None:
```

Comment:
These parameters are not required. The app name is the `name` parameter, and the section name is [here](https://github.com/snapcore/snapcraft/pull/1844/files#diff-c79d156505602fddd3f8b951fb63296cR46).